diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocFormatting.kt

```diff
@@ -148,8 +138,10 @@ class KdocFormatting(private val configRules: List<RulesConfig>) : Rule("kdoc-fo
             val hasSubject = tag.getSubjectName()?.isNotBlank() ?: false
             if (!hasSubject && tag.getContent().isBlank()) return@filter false
 
-            hasSubject && tag.node.findChildBefore(KDOC_TEXT, WHITE_SPACE)?.text != " "
-                || tag.node.findChildAfter(KDOC_TAG_NAME, WHITE_SPACE)?.text != " "
+            hasSubject && (tag.node.findChildBefore(KDOC_TEXT, WHITE_SPACE)?.text != " "
+                    && !(tag.node.findChildBefore(KDOC_TEXT, WHITE_SPACE)?.isWhiteSpaceWithNewline() ?: false))
+                    || (tag.node.findChildAfter(KDOC_TAG_NAME, WHITE_SPACE)?.text != " "
+                    && !(tag.node.findChildAfter(KDOC_TAG_NAME, WHITE_SPACE)?.isWhiteSpaceWithNewline() ?: false))
```

Comment:
My brain hurts when trying to understand it. Please declare a couple of additional boolean variables with parts of this complex expression.